### PR TITLE
Initial partially working GitHub Action

### PR DESCRIPTION
**Pull request by Jacalz**
### Description:

Missing  features are updating coverage, verifying go modules and not running vendor tests or mobile tests.

### Checklist:

- [ ] Tests included.
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.

#### Where applicable:

- [ ] Public APIs match existing style.
- [ ] Any breaking changes have a deprecation path or have been discussed.

### CODE REVIEW

File: .github/workflows/platform_tests.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+on: [push, pull_request]
+
+jobs:
+  platform_tests:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        go-version: [1.12.x, 1.14.x, 1.15.x]
+        os: [ubuntu-latest, windows-latest, macos-latest]
+
```
Comment (andydotxyz) [inline .github/workflows/platform_tests.yml:11]: To get just 1.12 legacy test run once the following might work:
```suggestion
    exclude:
      - go-version: 1.12.x
        os: [windows-latest, macos-latest]
```

Note: not tested!

Comment (Jacalz) [reply .github/workflows/platform_tests.yml:11]: Nice idea. Will give it a tests :)

**Comment by Jacalz** [comment]
Sorry for the force-push. Had applied the rebase incorrectly.

**Comment by Jacalz** [comment]
Tests seem to be failing due to issue with animation test. Pining @andydotxyz as a reminder.

**Comment by Jacalz** [comment]
Don't know what git is up to on my end. It wants me to force push all the time.

File: .github/workflows/platform_tests.yml (hunk 1 above)
Comment (Jacalz) [reply .github/workflows/platform_tests.yml:11]: As discussed earlier, we now run only on latest and newsest.

**Review by andydotxyz — APPROVED** [review]

**Pull request merged**